lucille-core/src/main/java/com/kmwllc/lucille/stage/DictionaryLookup.java

```diff
@@ -156,4 +193,40 @@ public Iterator<Document> processDocument(Document doc) throws StageException {
 
     return null;
   }
+
+  /**
+   * Get a Reader for the given path.
+   * todo consider moving this method to a utility class
+   *
+   * @param path file path
+   * @return Reader object
+   * @throws StageException if the file does not exist or cannot be read
+   */
+  private static Reader getFileReader(String path) throws StageException {
+    try {
+      return FileUtils.getReader(path);
+    } catch (NullPointerException | IOException e) {
+      throw new StageException("File does not exist: " + path);
+    }
+  }
+
+  /**
+   * Count the number of lines in a file.
+   * todo consider moving this method to a utility class
+   *
+   * @param filename file path
+   * @return number of lines
+   * @throws StageException if the file does not exist or cannot be read
+   */
+  private static int countLines(String filename) throws StageException {
+    try (BufferedReader reader = new BufferedReader(getFileReader(filename))) {
+      int lines = 0;
+      while (reader.readLine() != null) {
+        lines++;
+      }
+      return lines;
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
```

Comment:
these two methods are good ones to reuse
what is a good place to put them?

Reply:
For counting lines in a file, something like this is more concise:

`    try (Stream<String> fileStream = Files.lines(Paths.get(filename))) {
        int noOfLines = (int) fileStream.count();}`

You can put it in `com.kmwllc.lucille.util.FileUtils`

Reply:
moved to FileUtils. your version is more concise. however, we want a way to reuse the getReader method so that tests that rely on the following path are interpreted correctly `classpath:path/to/file`